hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestCacheDirectives.java

```diff
@@ -1575,7 +1575,7 @@ public void testNoLookupsWhenNotUsed() throws Exception {
     CacheManager cm = cluster.getNamesystem().getCacheManager();
     LocatedBlocks locations = Mockito.mock(LocatedBlocks.class);
     cm.setCachedLocations(locations);
-    Mockito.verifyZeroInteractions(locations);
```

Comment:
I am getting worn down by mockito changes *for no obvious reasons*

is there a way to add a MockitoSupport class to hadoop-common test and then have a static VerifyZeroInteractions() method? that way backporting test suites to an older version with a matching MockitoSupport implementation is straightforward

Reply:
``` java
  public static void verifyZeroInteractions(Object... mocks)
      throws InvocationTargetException, IllegalAccessException,
      NoSuchMethodException {
    String mockitoVersion = Mockito.class.getPackage().getSpecificationVersion();
    Method verifyInteractions;
    if (mockitoVersion.compareTo("4.1.1") < 0){
      verifyInteractions = Mockito.class.getMethod("verifyZeroInteractions",mocks.getClass());
    } else {
      verifyInteractions = Mockito.class.getMethod("verifyNoInteractions",mocks.getClass());
    }
    verifyInteractions.invoke(null,mocks);
  }
```
Not tried this but something like this could work. But is it okay to use reflections ?

Reply:
Or maybe you are suggesting to only include a static method by that name and inside that call the noInteractions method so that on backport, the changes required would be minimal?

Reply:
exactly! do a somple wrapper to the new lib and on a backport one class to change

Reply:
still hoping to see this to assist backports